File: client/lib/models/player.dart
class Player {
  final String nickName;
  final String socketId;
  final int points;
  final PlayerType type;

  Player({
    required this.nickName,
    required this.socketId,
    required this.points,
    required this.type,
  });

  Player.fromJson(Map<String, dynamic> json)
      : nickName = json['nickName'],
        socketId = json['socketID'],
        points = json['points'],
        type = json['playerType'] == 'x' ? PlayerType.x : PlayerType.o;

  Map<String, dynamic> toJson() {
    return {
      'nickName': nickName,
      'socketID': socketId,
      'points': points,
      'playerType': type == PlayerType.x ? 'x' : 'o'
    };
  }
}

enum PlayerType { x, o }

File: server/model/player.js
const mongoose = require('mongoose')

const playerSchema = new mongoose.Schema({
    nickName: {
        type: String,
        required: [true, 'please provide nickname'],
        trim: true
    },
    socketID: {
        type: String,
    },
    points: {
        type: Number,
        default: 0
    },
    playerType: {
        required :[true, 'please provide type'],
        type: String,
        enum: ['x', 'o']
    }
});

module.exports = playerSchema